src/3rdparty/md5/md5.h

```diff
@@ -53,6 +53,25 @@
 #ifndef MD5_INCLUDED
 #define MD5_INCLUDED
 
+/** The number of bytes in a MD5 hash. */
+static const size_t MD5_HASH_BYTES = 16;
+
+/**
+ * Container for storing a MD5 hash/checksum/digest.
+ */
+struct MD5Hash : public std::array<byte, MD5_HASH_BYTES> {
```

Comment:
Strictly speaking, you wouldn't even need a class, but one can definitely argue if the code below is in fact better or worse:
```c++
using MD5 = std::array<byte, MD5_HASH_BYTES>;

inline MD5 &operator^=(MD5 &rhs, const MD5 &lhs)
{
    for (size_t i = 0; i < rhs.size(); i++) rhs[i] ^= lhs[i];
    return rhs;
}
````

Reply:
I think that is neater and more extensible -- no extra class indeed.